pkg/kv/store.go

```diff
@@ -0,0 +1,97 @@
+package kv
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"sync"
+)
+
+var (
+	ErrUnknownDriver   = errors.New("unknown driver")
+	ErrOperationFailed = errors.New("operation failed")
+	ErrMissingKey      = errors.New("missing key")
+	ErrMissingValue    = errors.New("missing value")
+	ErrNotFound        = errors.New("not found")
+	ErrClosedEntries   = errors.New("closed entries")
+)
+
+type Entry struct {
+	Key   []byte
```

Comment:
Let's have a concrete `Key` & `Value` types to avoid mixup by the caller or the implementor.

Reply:
And `ValuePredicate`?

Reply:
Went over several libraries (key/value databases related) - all uses native types as key/value - it is string/[]byte for key and []byte for value:
https://github.com/dgraph-io/badger/blob/master/db.go
https://github.com/etcd-io/bbolt/blob/master/bucket.go#L261
https://github.com/recoilme/pudge/blob/master/api.go
https://git.mills.io/prologic/bitcask/src/branch/master/bitcask.go#L121
https://github.com/peterbourgon/diskv/blob/master/index.go